### PR TITLE
V0.4.0

### PR DESCRIPTION
**Pull request by liquorvicar**
- Add custom CS ruleset, including linting of test files
- Add support for objects with multiple properties
- Add support for required and optional properties

### CODE REVIEW

File: src/Properties/EnumProperty.php [hunk 1]
```diff
@@ -61,7 +61,7 @@ public function addTo(ClassType $class)
     {
         $class->addProperty($this->name)
             ->setVisibility('private')
-            ->addComment("@var $this->enumName");
+            ->addComment("@var string");
```
Comment (johnwards) [inline src/Properties/EnumProperty.php]: Why did you need to change this?

Comment (liquorvicar) [reply src/Properties/EnumProperty.php]: Because it was wrong. With enums (which are assumed to be made up of string values for now), we grab the value out of the enum in the constructor, see `\Elsevier\JSONSchemaPHPGenerator\Properties\EnumProperty::constructorBody` (https://github.com/elsevier-io/json-schema-php-generator/blob/master/src/Properties/EnumProperty.php#L73)
